src/pubsub.c

```diff
@@ -538,9 +533,29 @@ int pubsubPublishMessage(robj *channel, robj *message, int sharded) {
  * Pubsub commands implementation
  *----------------------------------------------------------------------------*/
 
+void addPubSubChannel(client *c, pubsubtype type) {
+    int j;
+    struct ClientFlags old_flags = c->flag;
+    c->flag.pushing = 1;
+    int number = (c->argc - 1) * 3;
+
+    if (c->resp == 2)
+        addReply(c, shared.mbulkhdr[number]);
+    else
+        addReplyPushLen(c, number);
```

Comment:
So with RESP3, this is still no normal in-band replies and one push message (instead of one per channel). This is still the main problem of `[UN][P|S]SUBSCRIBE` IMO.

To fix this behaviour, SUBSCRIBE should return an in-band reply. Either the list of channels as an array or map-reply, or just +OK and send the channels as push.

Reply:
I do not want to import this change in Valkey 8.0. I wish we could do this change on Valkey 8.X or 9 version. Error need to be correct, never late ^_^

Reply:
I think we cannot fix it ever. Only maybe with a CLIENT CAPA or RESP4.

Reply:
Yeah, I agree with this being a potentially significant breaking change. I think maybe CLIENT CAPA could fix it (or RESP4).

Reply:
I update the design, add a new parameter for client capa command, but the parameter name is just temporarily. If you have better name, please let me know or we can discuss it in the meeting